core/commonMain/src/kotlinx/serialization/encoding/Encoding.kt

```diff
@@ -473,16 +473,41 @@ public interface CompositeEncoder {
 /**
  * Begins a structure, encodes it using the given [block] and ends it.
  */
-public inline fun Encoder.encodeStructure(descriptor: SerialDescriptor, block: CompositeEncoder.() -> Unit) {
-    val composite = beginStructure(descriptor)
-    var ex: Throwable? = null
-    try {
-        composite.block()
-    } catch (e: Throwable) {
-        ex = e
-        throw e
-    } finally {
-        // End structure only if there is no exception, otherwise it can be swallowed
-        if (ex == null) composite.endStructure(descriptor)
+public inline fun Encoder.encodeStructure(
```

Comment:
Sorry if I wasn't clear in my message, I've meant only new signatures (`encodeCollection` overloads).

With the change in the existing signature, I cannot merge it as is, as it will be breaking change and will prevent us from releasing 1.3.2.

Reply:
Ah, fixed.